### PR TITLE
Add Cloud Task Queue resource

### PR DESCRIPTION
**Pull request by mccall114**
New resource `google_task_queue` as part of issue https://github.com/terraform-providers/terraform-provider-google/issues/2367.

First time contributor here. I read the guidelines about splitting out new packages in a separate PR, and I'm happy to do that if this one is ok.

### CODE REVIEW

**Comment by mccall114** [comment]
Linter failed in travis. Running locally with go 1.11.13 confirms that it's failing on code I didn't touch.

File: google/resource_task_queue.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+		return fmt.Errorf("Error deleting task queue: %s", err)
+	}
+
+	log.Printf("[DEBUG] Finished deleting Task Queue: %#v\n%#v\n", name, resp)
+
+	return nil
+}
+
+func resourceTaskQueueImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
```
Comment (bamItsCam) [inline google/resource_task_queue.go]: Consider doing what resource_kms_key_ring.go does for its import, using parseImportId() and replaceVars().

File: google/resource_task_queue.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+				},
+			},
+			"rate_limits": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_burst_size": {
```
Comment (tysen) [inline google/resource_task_queue.go]: The API docs say this field is output only so we can omit it from the schema.

File: google/resource_task_queue.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+						},
+						"max_concurrent_dispatches": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"retry": {
```
Comment (tysen) [inline google/resource_task_queue.go]: I would prefer if this matched the API: `retry_config`

File: google/resource_task_queue.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+				ForceNew: true,
+			},
+			"app_engine_routing_override": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host": {
```
Comment (tysen) [inline google/resource_task_queue.go]: Also output only.

File: google/resource_task_queue_test.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+		CheckDestroy: testAccTaskQueueDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTaskQueue_basic(queueName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTaskQueueExists(
+						"google_task_queue.fizzbuzz", location, queueName, &queue),
+					testAccCheckTaskQueueEquals(
+						"google_task_queue.fizzbuzz", &queue),
+				),
```
Comment (tysen) [inline google/resource_task_queue_test.go]: I don't think we need these checks (nor the funcs, and I think that also means we can avoid vendoring this new `context` dependency) - the `ImportStateVerify` is sufficient.

**Comment by mccall114** [comment]
@tysen, I've removed those tests and renamed `retry` to `retry_config`. I left `max_burst_size` and `host` so that we could still see them in the output. LMK what you think.

File: google/resource_task_queue_test.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_task_queue" {
+			continue
+		}
+
+		attributes := rs.Primary.Attributes
+
+		expectedName := fmt.Sprintf("projects/%s/locations/%s/queues/%s", config.Project, attributes["location"], attributes["name"])
+
+		_, err := config.clientCloudTasks.Projects.Locations.Queues.Get(expectedName).Context(context.Background()).Do()
```
Comment (tysen) [inline google/resource_task_queue_test.go]: Why include this call to `Context`? Does this not work without it like the other services?

**Comment by jba** [comment]
I'd love to start using this. What is holding it up? Can I help?

**Comment by tysen** [comment]
Still waiting for a response to the question in my review.

**Comment by pokojskis** [comment]
I would really love to use it

File: google/resource_task_queue_test.go (hunk 6 above)
Comment (mccall114) [reply google/resource_task_queue_test.go]: Sorry for the delay, I was on vacation. I've removed the call to `Context`.

**Comment by tysen** [comment]
PR looks good now. If you could submit separate PRs (one for TPG and one for TPGB) to vendor the new dependency (and then rebase this PR to remove the vendoring here), afterwards I can upstream this PR into MM and get it merged.

**Comment by mccall114** [comment]
@tysen, I've rebased this branch and opened https://github.com/terraform-providers/terraform-provider-google/pull/4781
and https://github.com/terraform-providers/terraform-provider-google-beta/pull/1315.

**Comment by tysen** [comment]
@mccall114 Looks like you need to sign the CLA https://github.com/GoogleCloudPlatform/magic-modules/pull/2587

**Comment by mccall114** [comment]
@tysen I just signed it

**Comment by tysen** [comment]
@mccall114 Thanks, will you post `@googlebot I fixed it.` on that PR?

**Comment by seriousManual** [comment]
any updates on this?

**Comment by tysen** [comment]
I had some issues finalizing this implementation, and it'll be easier to maintain and extend if it's generated by Magic Modules, so I submitted https://github.com/GoogleCloudPlatform/magic-modules/pull/2662 which is under review.

**Comment by tysen** [comment]
Should be in the next release: https://github.com/GoogleCloudPlatform/magic-modules/pull/2662

**Comment by mccall114** [comment]
Thanks @tysen!

**Pull request closed without merging**